src/lib/viewers/ResultsViewer.ts

```diff
@@ -0,0 +1,136 @@
+import {RunResults, SeverityLevel, Violation} from '@salesforce/code-analyzer-core';
+import {Display} from '../Display';
+import {BundleName, getMessage} from '../messages';
+
+export interface ResultsViewer {
+	view(results: RunResults): void;
+}
+
+abstract class AbstractResultsViewer implements ResultsViewer {
+	protected display: Display;
+
+	public constructor(display: Display) {
+		this.display = display;
+	}
+
+	public abstract view(results: RunResults): void;
+}
+
+export class ResultsDetailViewer extends AbstractResultsViewer {
+	/**
+	 * The display format consumes a lot of vertical space. As such, we're choosing to limit the total
+	 * number of violations that it will display before truncating the results and instructing the user
+	 * to consult one of the output files.
+	 */
+	public static readonly RESULTS_CUTOFF = 10;
```

Comment:
My personal gut feeling is that we shouldn't have any cutoff for the detail viewer... and if we did, then make the default to not have a cutoff (or at least make the default much larger). Forcing the user to have to output to a file to see all the results seems like we are preventing some ease-of-use cases. But that is just my opinion. If anything, having a cutoff on the table (non-detail) view would make more sense... since users can't expect us to show everything in the table anyway (since messages need to already be truncated).

Reply:
I litigated this with John and Marcelino on Quip, but we can revisit it if we want to.